WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt

```diff
@@ -226,60 +227,83 @@ class MySiteViewModel @Inject constructor(
+                        onPostItemClick = this::onPostItemClick,
+                        onFooterLinkClick = this::onPostCardFooterLinkClick
+                ),
+                QuickActionsCardBuilderParams(
+                        siteModel = site,
+                        activeTask = activeTask,
+                        onQuickActionStatsClick = this::quickActionStatsClick,
+                        onQuickActionPagesClick = this::quickActionPagesClick,
+                        onQuickActionPostsClick = this::quickActionPostsClick,
+                        onQuickActionMediaClick = this::quickActionMediaClick
+                ),
+                QuickStartCardBuilderParams(
+                        quickStartCategories = quickStartCategories,
+                        onQuickStartBlockRemoveMenuItemClick = this::onQuickStartBlockRemoveMenuItemClick,
+                        onQuickStartTaskTypeItemClick = this::onQuickStartTaskTypeItemClick
+                ),
+                SiteInfoCardBuilderParams(
+                        site = site,
+                        showSiteIconProgressBar = showSiteIconProgressBar,
+                        titleClick = this::titleClick,
+                        iconClick = this::iconClick,
+                        urlClick = this::urlClick,
+                        switchSiteClick = this::switchSiteClick,
+                        activeTask = activeTask
+                )
+        )
+        val dynamicCards = dynamicCardsBuilder.build(
+                quickStartCategories,
+                pinnedDynamicCard,
+                visibleDynamicCards,
+                this::onDynamicCardMoreClick,
+                this::onQuickStartTaskCardClick
+        )
+
+        val siteItems = siteItemsBuilder.build(
+                SiteItemsBuilderParams(
+                        site = site,
+                        activeTask = activeTask,
+                        backupAvailable = backupAvailable,
+                        scanAvailable = scanAvailable,
+                        onClick = this::onItemClick
+                )
+        )
+        return orderForDisplay(cards, dynamicCards, siteItems)
+    }
 
     private fun buildNoSiteState(): NoSites {
         // Hide actionable empty view image when screen height is under specified min height.
         val shouldShowImage = displayUtilsWrapper.getDisplayPixelHeight() >= MIN_DISPLAY_PX_HEIGHT_NO_SITE_IMAGE
         return NoSites(shouldShowImage)
     }
 
+    private fun orderForDisplay(
+        cards: List<MySiteCardAndItem>,
+        dynamicCards: List<MySiteCardAndItem>,
+        siteItems: List<MySiteCardAndItem>
+    ): List<MySiteCardAndItem> {
+        val indexOfPostsCard = cards.indexOfFirst { it is PostCard }
+        return if (indexOfPostsCard == -1) {
+            cards + dynamicCards + siteItems
+        } else {
+            return mutableListOf<MySiteCardAndItem>().apply {
```

Comment:
Nitpick: `return` keyword can be dropped at this line as we already have it before the `if-else` block.